### DISCUSSION TITLE
Difference between JunctionOneToOne1Phase and VolumeJunction1Phase

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#27090] [Q&A Modules: Thermal Hydraulics] [answered]
Hi all,

I would like to know the difference between `JunctionOneToOne1Phase` and `VolumeJunction1Phase` in THM.

If the simulation is Pipe1-Junction-Pipe2, I think the result using these two junction types should be the same. But I tested it using a lax_shock_tube test, the result is not the same. What's the reason?

[lax_shock_tube_OnetoOne.txt](https://github.com/idaholab/moose/files/14603113/lax_shock_tube_OnetoOne.txt)
[lax_shock_tube_VolumeBranch.txt](https://github.com/idaholab/moose/files/14603119/lax_shock_tube_VolumeBranch.txt)

Result:
![image](https://github.com/idaholab/moose/assets/118275988/5601b248-a168-42e4-9598-719ca3c633a2)

**Comment by GiudGiud** [comment]
@joshuahansel 
it looks to me like the junctions arent exactly at the same locations

**Reply by SomnusYu** [reply]
It is truly at the same location, please see the input file. The discontinuity of JunctionOneToOne1Phase's result is caused by the shock tube, not the junction.

**Comment by joshuahansel** [comment, marked as answer]
One difference between the two is that the volume junction doesn't do any slope reconstruction across the junction (maybe it could after some development? 🤔 ). So first compare them both with `rdg_slope_reconstruction = none`. The differences in the results are quite striking though, so I'd be surprised if that were everything.

I think the main reason may be the volume you've chosen does not yield an equivalent problem. You have `A = 1` and `volume = 1`, so it's like you're junction is actually `L = volume / A = 1` long, twice the length of each of the pipes. For an equivalent problem, probably choose the volume to give you an element length. Then take away one element from one of the connected pipes. Then you probably need to shift the ICs by a half element. Make sense? You basically want to replace one of the elements from the one-to-one junction case with this volume junction.

**Reply by SomnusYu** [reply]
I got it! Thanks a lot! I use `rdg_slope_reconstruction = none` for these two cases, but it still has a little difference. What's the reason?
![image](https://github.com/idaholab/moose/assets/118275988/5246f6a6-6841-4371-af76-63623d118667)

**Reply by joshuahansel** [reply]
And did you try what I said in the second paragraph too? If so, please supply input files again.

**Reply by SomnusYu** [reply]
Yes, I changed volume=_element_lenth * _A = 0.01. Here are the input files.
[lax_shock_tube_VolumeBranch.txt](https://github.com/idaholab/moose/files/14616610/lax_shock_tube_VolumeBranch.txt)
[lax_shock_tube_OnetoOne.txt](https://github.com/idaholab/moose/files/14616611/lax_shock_tube_OnetoOne.txt)

**Reply by joshuahansel** [reply]
It appears you didn't do the element deletion I talked about: For your volume junction input file, shorten the length and number of elements of the right channel by 1 element.

**Reply by joshuahansel** [reply]
What's happening is that your volume junction case has additional length that is not captured in your plot.

**Reply by SomnusYu** [reply]
I understand it! Thanks a lot!
